src/audio/host.c

```diff
@@ -221,7 +221,7 @@ static void host_dma_cb(void *arg, enum notify_id type, void *data)
 	struct host_data *hd = comp_get_drvdata(dev);
 	uint32_t bytes = next->elem.size;
 
-	comp_cl_dbg(&comp_host, "host_dma_cb() %p", (uintptr_t)&comp_host);
+	comp_cl_dbg(&comp_host, "host_dma_cb() %p", &comp_host);
```

Comment:
So Zephyr tracing system actually supports pointers directly?

Reply:
Yes, is %p supposed to print a pointer?

Reply:
Yes, %p is intended to print a pointer. The cast is solely due to how "cheap tracing" (my naming, no clue if there is any "official" name) is implemented that it supports up to 4 parameters, all integer (and in all current platforms, all 32-bit integers). Every other type of parameter must fit within that integer thing. Pointers work. Small integers also work due to how smaller integers turn into the "int" type when passed to vararg so it's also fine.

It'll be nice when this becomes the main thing, as simple things you don't typically think about much, like logging, become just a tad bit less confusing.

Reply:
So is the solution OK? Otherwise you need to change %p to whatever uintptr_t type is.

Reply:
This one is ok, yes indeed we have pointers here. I just inserted (uintptr_t) casts because passing pointers without casts in the current SOF logging system is not possible. I was just asking about it because I got used to the limited system.